build/config/pom.xml

```diff
@@ -121,6 +126,125 @@
+                            <mainClass>org.hibernate.checkstyle.report.AnnotationReportGenerator</mainClass>
+                            <arguments>
+                                <!-- The location of classes to process: -->
+                                <argument>${tmpdir.dependencies-to-parse}/hibernate-core</argument>
+                                <!--
+                                    The report output directory path.
+                                    We write to the build output directory so that the report is included in the jar:
+                                -->
+                                <argument>${project.build.outputDirectory}</argument>
+                                <!-- The report file name main part. Generator will add any suffixes and extensions it needs to it. -->
+                                <argument>hibernate-core-incubating</argument>
+                                <!-- The annotation to look for: -->
+                                <argument>org.hibernate.Incubating</argument>
+                                <!-- Ignore rules: -->
+                                <!--
+                                    Rules can start with a `-Rinternal ` or a `-Rpublic ` prefixes.
+                                    This would mean that a corresponding pattern will only be applied to filtering of internal/public rules.
+                                    If no prefix is specified then rule is considered to be both public and internal at the same time.
+                                -->
+                                <argument>^org\.hibernate\.query\.Query$</argument>
+                                <argument>^org\.hibernate\.query\.SelectionQuery$</argument>
+                                <argument>^org\.hibernate\.query\.MutationQuery$</argument>
+                                <argument>^org\.hibernate\.query\.BindableType$</argument>
+
+                                <argument>^org\.hibernate\.metamodel\.MappingMetamodel$</argument>
+                                <argument>^org\.hibernate\.metamodel\.mapping\.Bindable$</argument>
+                                <argument>^org\.hibernate\.metamodel\.mapping\.SelectableMapping$</argument>
+                                <argument>^org\.hibernate\.metamodel\.mapping\.SelectablePath$</argument>
+                                <argument>^org\.hibernate\.metamodel\.model\.domain\.JpaMetamodel$</argument>
+
+                                <argument>^org\.hibernate\.boot\.spi\.BootstrapContext$</argument>
+                                <argument>^org\.hibernate\.collection\.spi\.PersistentCollection$</argument>
+                                <argument>^org\.hibernate\.type\.spi\.TypeConfiguration$</argument>
+                                <argument>^org\.hibernate\.query\.spi\.ScrollableResultsImplementor$</argument>
+                                <argument>^org\.hibernate\.query\.sqm\.spi\.SqmCreationContext$</argument>
```

Comment:
All of these make sense. Though I'd expect us to use `@SuppressForbiddenApis` to add exceptions instead, like we've done in other cases.